django_geosource/models.py

```diff
@@ -129,26 +131,39 @@ def refresh_data(self):
             layer = self.get_layer()
             begin_date = datetime.now()
             row_count = 0
+            total = 0
 
-            for row in self._get_records():
+            for i, row in enumerate(self._get_records()):
+                total += 1
                 geometry = row.pop(self.SOURCE_GEOM_ATTRIBUTE)
                 try:
                     identifier = row[self.id_field]
                 except KeyError:
-                    raise Exception(
-                        "Can't find identifier field in one or more records"
+                    msg = "Can't find identifier field for this record"
+                    self.report["status"] = "Warning"
+                    self.report.setdefault("message", []).append(msg)
+                    self.report.setdefault("lines", {}).setdefault(f"{i}", []).append(
+                        msg
                     )
+                    self.save(update_fields=["report"])
```

Comment:
Could be very time consuming, just construct the report object, then save it.

Reply:
Done.